homeassistant/backports/enum.py

```diff
@@ -1,13 +1,23 @@
 """Enum backports from standard lib."""
 from __future__ import annotations
 
-from enum import Enum
+from enum import Enum, EnumMeta
 from typing import Any, TypeVar
 
 T = TypeVar("T", bound="StrEnum")
 
 
-class StrEnum(str, Enum):
+class StrEnumMeta(EnumMeta):
+    """Use by StrEnum to match strings."""
+
+    def __contains__(cls, member: object) -> bool:
+        """Override to allow string comparison."""
+        if isinstance(member, str):
+            return member in cls._value2member_map_
+        return super().__contains__(member)
```

Comment:
This would solve the issue of checking if a string is in an `Enum`.
But then, since we are always checking against a registry_entry maybe #62225 would make this redundant and should be completed first?

cc @scop / @davet2001

Reply:
Note, this the base implementation in `EnumMeta`:

```python
    def __contains__(cls, member):
        if not isinstance(member, Enum):
            raise TypeError(
                "unsupported operand type(s) for 'in': '%s' and '%s'" % (
                    type(member).__qualname__, cls.__class__.__qualname__))
        return isinstance(member, cls) and member._name_ in cls._member_map_

```

Reply:
I've moved it out of backports as that is no longer a backport...

Reply:
I thought some more about this, and I think that we should merge #64618 before we progress with this PR